digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -952,8 +952,9 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     */
   private def setBlackUnderline(addresses: Seq[RoadAddressLink]): Seq[RoadAddressLink] = {
     time(logger, "Set the black underline") {
-      val (streetRoads, othersRoads) = addresses.partition(_.roadType == RoadType.MunicipalityStreetRoad)
-      streetRoads.map(_.copy(blackUnderline = true)) ++ othersRoads
+      val typesForBlackUnderline = Set(RoadType.MunicipalityStreetRoad.value, RoadType.PrivateRoadType.value)
+      val (roadsNeedingUnderline, otherRoads) = addresses.partition(a =>typesForBlackUnderline.contains(a.roadType.value))
+      roadsNeedingUnderline.map(_.copy(blackUnderline = true)) ++ otherRoads
```

Comment:
Rows 956 and 957 could be replaced with 
`addresses.map(a => a.copy(blackUnderline = typesForBlackUnderline.contains(a.roadType.value)))`